src/core/CoreModuleDirector.ts

```diff
@@ -291,35 +333,63 @@ export class CoreModuleDirector {
   }
 
   public getSubscriptionOfTypeWithToken(
-    type: ModelName,
+    type: SubscriptionChannel | undefined,
     token: string,
   ): OSModel<SupportedSubscription> | undefined {
     logMethodCall('CoreModuleDirector.getSubscriptionOfTypeWithToken', {
       type,
       token,
     });
+
+    let subscriptions: Record<string, OSModel<SupportedSubscription>>;
+
     switch (type) {
-      case ModelName.EmailSubscriptions: {
-        const emailSubscriptions = this.getEmailSubscriptionModels();
-        return Object.values(emailSubscriptions).find(
-          (subscription) => subscription.data.token === token,
-        );
-      }
-      case ModelName.SmsSubscriptions: {
-        const smsSubscriptions = this.getSmsSubscriptionModels();
-        return Object.values(smsSubscriptions).find(
-          (subscription) => subscription.data.token === token,
-        );
-      }
-      case ModelName.PushSubscriptions: {
-        const pushSubscriptions = this.getAllPushSubscriptionModels();
-        return Object.values(pushSubscriptions).find(
-          (subscription) => subscription.data.token === token,
-        );
-      }
+      case SubscriptionChannel.Email:
+        subscriptions = this.getEmailSubscriptionModels();
+        break;
+      case SubscriptionChannel.SMS:
+        subscriptions = this.getSmsSubscriptionModels();
+        break;
+      case SubscriptionChannel.Push:
+        subscriptions = this.getAllPushSubscriptionModels();
+        break;
       default:
         return undefined;
     }
+
+    return Object.values(subscriptions).find(
+      (subscription) => subscription.data.token === token,
+    );
+  }
+
+  /**
+   * Helper that checks if a given SubscriptionType is a push subscription.
+   */
+  public isPushSubscriptionType(type: SubscriptionType): boolean {
+    switch (type) {
+      case SubscriptionType.ChromePush:
+      case SubscriptionType.SafariPush:
+      case SubscriptionType.SafariLegacyPush:
+      case SubscriptionType.FirefoxPush:
+        return true;
+      default:
+        return false;
+    }
+  }
```

Comment:
This method should probably be moved elsewhere like `SubscriptionModels.ts`

Reply:
I moved both methods into `SubscriptionHelper.ts` https://github.com/OneSignal/OneSignal-Website-SDK/pull/1179/commits/d4bbe89a7513745d8bf7bb0c9efb1464bd31c361